src/Microsoft.OpenApi/Models/OpenApiDocument.cs

```diff
@@ -112,19 +118,36 @@ public void SerializeAsV3(IOpenApiWriter writer)
             writer.WriteStartObject();
 
             // openapi
-            writer.WriteProperty(OpenApiConstants.OpenApi, "3.0.1");
-
+            switch (version)
+            {
+                case OpenApiSpecVersion.OpenApi3_1:
+                    writer.WriteProperty(OpenApiConstants.OpenApi, "3.1.0");
+                    break;
+                case OpenApiSpecVersion.OpenApi3_0:
+                    writer.WriteProperty(OpenApiConstants.OpenApi, "3.0.1");
+                    break;
+                default:
+                    writer.WriteProperty(OpenApiConstants.OpenApi, "3.0.1");
+                    break;
+            }
+            
             // info
             writer.WriteRequiredObject(OpenApiConstants.Info, Info, (w, i) => i.SerializeAsV3(w));
 
+            // jsonSchemaDialect
+            if(version == OpenApiSpecVersion.OpenApi3_1)
+                writer.WriteProperty(OpenApiConstants.JsonSchemaDialect, JsonSchemaDialect);
+
             // servers
             writer.WriteOptionalCollection(OpenApiConstants.Servers, Servers, (w, s) => s.SerializeAsV3(w));
 
             // paths
             writer.WriteRequiredObject(OpenApiConstants.Paths, Paths, (w, p) => p.SerializeAsV3(w));
 
             // webhooks
-            writer.WriteOptionalMap(
+            if (version == OpenApiSpecVersion.OpenApi3_1)
```

Comment:
We should try and avoid these kinds of conditional statements if possible.  We could refactor the common code into a common method and share it across the v3.1 and v3.0 serialization.  We don't need to worry about the order of properties.

Reply:
It seems like the order of properties changes when we use this approach. Here is a snippet of the resulting document after refactoring to use a common method:

![image](https://user-images.githubusercontent.com/36787645/218049748-b7af0813-2710-4282-af00-2da91941a5b0.png)

Result:
```
    openapi: '3.1.0'
    info:
      title: JsonSchemaDialectTest
      version: 1.0.0
    paths: { }
    jsonSchemaDialect: http://json-schema.org/draft-07/schema#
```